### PR TITLE
Refactor useForm test

### PR DESCRIPTION
**Pull request by keiya01**

### CODE REVIEW

File: src/useForm.test.tsx [hunk 1]
```diff
@@ -76,67 +76,30 @@ describe('useForm', () => {
...
-            name: 'test',
-            type: 'radio',
-          },
-        },
-        isRadioOrCheckbox: true,
-        handleChange: expect.any(Function),
-      });
-      expect(onDomRemove).toBeCalled();
-    });
+    test.each([['text'], ['radio'], ['checkbox']])(
```
Comment (bluebill1049) [inline src/useForm.test.tsx:79]: nice

File: src/useForm.test.tsx [hunk 2]
```diff
@@ -442,37 +409,81 @@ describe('useForm', () => {
...
+    const tests: [
+      string,
+      [string | string[] | undefined, unknown | undefined, string | undefined],
+      {
+        watchValue?: unknown;
+        watchFieldsHookRef: string[];
+        watchFieldsRef: string[];
+        isWatchAllRef: boolean;
+      },
+    ][] = [
```
Comment (bluebill1049) [inline src/useForm.test.tsx:421]: a bit hard to read this one, but smart. maybe we can improve a bit on the readability.

**Review by bluebill1049 — APPROVED** [review]
awesome improvement.

File: src/useForm.test.tsx (hunk 2 above)
Comment (keiya01) [reply src/useForm.test.tsx:421]: Good point. I will try to improve this test.
Thank you.

**Pull request merged**